lld/COFF/Driver.cpp

```diff
@@ -365,6 +380,24 @@ bool LinkerDriver::isDecorated(StringRef sym) {
          (!ctx.config.mingw && sym.contains('@'));
 }
 
+static LLVM_THREAD_LOCAL bool executingFirstQueue;
+
+static bool executeQueue(std::list<std::function<void()>> &queue) {
+  bool didWork = !queue.empty();
+  while (!queue.empty()) {
+    queue.front()();
+    queue.pop_front();
+  }
+  return didWork;
+}
+
+static bool executeFirstQueue(std::list<std::function<void()>> &queue) {
```

Comment:
Comments here about the purpose of the first queue (to load only .objs from the command line) would be helpful.